### PR TITLE
Implement the Numbering class for the NumberingDefinitionsPart

### PR DESCRIPTION
**Pull request by kylegibson**
https://msdn.microsoft.com/en-us/library/documentformat.openxml.wordprocessing.numbering%28v=office.14%29.aspx

Depends on #128

### CODE REVIEW

File: pydocx/openxml/packaging/numbering_definitions_part.py [hunk 1]
```diff
@@ -20,3 +21,14 @@ class NumberingDefinitionsPart(OpenXmlPart):
...
+    def __init__(self, *args, **kwargs):
+        super(NumberingDefinitionsPart, self).__init__(*args, **kwargs)
+        self._numbering = None
+
+    @property
+    def numbering(self):
+        if self._numbering:
+            return self._numbering
+        self._numbering = Numbering.load(self.root_element)
+        return self._numbering
```
Comment (kylegibson) [inline pydocx/openxml/packaging/numbering_definitions_part.py]: I need to add a test on this

File: tests/openxml/wordprocessing/test_level.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        self.assertEqual(level.paragraph_style, "normal")
+
+    def test_run_properties_child(self):
+        xml = b'''
+            <lvl>
+                <rPr />
+            </lvl>
+        '''
+        level = self._load_from_xml(xml)
+        self.assertEqual(level.run_properties.__class__, RunProperties)
```
Comment (jlward) [inline tests/openxml/wordprocessing/test_level.py]: Comparing classes in this way is not the correct way of doing it. You should be using `isinstance` instead. For all cases in this diff.

**Comment by jlward** [comment]
After you update the test cases to use `isinstance`, this looks good.

File: tests/openxml/wordprocessing/test_level.py (hunk 2 above)
Comment (kylegibson) [inline tests/openxml/wordprocessing/test_level.py]: Could you help me understand why it's not the correct way, and why `isinstance` is preferred in this particular case, and all cases in this diff?

If I was doing control flow based on the type of an object I would use `isinstance` (for example, if obj is a string, do X) but in this case I am specifically wanting to ensure the class is what it should be.

Also, the benefit of doing it this way is that when the comparison fails, the assertion shows what the actual class is. I don't get that with `isinstance`.

Comment (jlward) [inline tests/openxml/wordprocessing/test_level.py]: http://www.siafoo.net/article/56

Comment (jlward) [inline tests/openxml/wordprocessing/test_level.py]: https://www.python.org/dev/peps/pep-0008/

Comment (kylegibson) [inline tests/openxml/wordprocessing/test_level.py]: The first resource makes a solid case that `isinstance` is preferred to `__class__` *when you are writing control flow logic*. Totally agree. I'm not doing that here. 

Also, as confirmed by the first resource, `isinstance` would return true if the class is a sub-class. In the model implementation schema, I feel that would be a bug. The class should not be a sub-class (of `RunProperties` in this case), but the exact class itself.

The PEP8 resource doesn't mention `__class__` at all. Instead, it says you should not use `type` in a conditional. I'm not using type, and I don't have a conditional. It doesn't say that `__class__` is to be avoided.

Just so we're on the same page, the alternative we're discussing is the following:

```python
assert isinstance(level.run_properties, RunProperties)
```

Downsides:

* When the assertion fails, there's no visibility to what the actual class is, just what it is not
* Captures sub-classes, which may not be desired (I feel they are not)

Upsides:

* Avoids having to access a dunder property
* This is how type checking is preferred for control flow

Thoughts?

Comment (jlward) [inline tests/openxml/wordprocessing/test_level.py]: > When the assertion fails, there's no visibility to what the actual class is, just what it is not

You could do something like

```python
assert isinstnace(level.run_properties, RunProperties), level.run_properties
```

Or even make a helper method that does this for you. That way, if your __repr__ method on the classes are set up correctly, you will know what class it actually is.

>  ... and I don't have a conditional

What do you think self.assertEqual uses behind the scenes? :) (Hint, it's a conditional)

Comment (kylegibson) [inline tests/openxml/wordprocessing/test_level.py]: How would I solve the sub-class issue? `insinstance` would return true for a sub-class. Wouldn't that be a bug?

Comment (jlward) [inline tests/openxml/wordprocessing/test_level.py]: Is it a bug? I don't think it is. sub classes are should be fine.

**Pull request merged**